apollo-router-core/src/plugin.rs

```diff
@@ -64,41 +64,59 @@ pub fn plugins() -> HashMap<String, PluginFactory> {
     PLUGIN_REGISTRY.lock().expect("Lock poisoned").clone()
 }
 
+/// All router plugins must implement the Plugin trait. This trait defines lifecycle hooks that enable hooking into Apollo Router services.
+/// The trait also provides a default implementations for each hook, which returns the associated service unmodified.
```

Comment:
Add a link to the online docs: https://www.apollographql.com/docs/router/customizations/native/#plugin-lifecycle